### DISCUSSION TITLE
/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /home/linuxbrew/.linuxbrew/lib/libsrt.so.1.4)

### DISCUSSION THREAD

**Discussion by jintor** [Homebrew/discussions#2320] [Linux] [answered]
CONTEXT : linuxbrew

WHAT IS WORKING : brew install imagemagick
magick -version : ImageMagick 7.1.0-10 Q16-HDRI x86_64 2021-10-10 https://imagemagick.org.......

PROBLEM : ffmpeg -i = /lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /home/linuxbrew/.linuxbrew/lib/libsrt.so.1.4)

WHAT I TRIED : brew cleanup + brew doctor + brew reinstall ffmpeg

and still get this error (all othe brew install are all ok execpt ffmpeg)

QUESTION : How can this be fixed ?

MY LINUX : AlmaLinux 8.4 Kernel: Linux 4.18.0-305.19.1.el8_4.x86_64

**Comment by SMillerDev** [comment]
What do `brew doctor` and `brew config` say?

**Reply by jintor** [reply]
strings /usr/lib64/libstdc++.so.6 | grep GLIBCXX ===>> .....GLIBCXX_3.4.25

brew config
HOMEBREW_VERSION: 3.2.17
ORIGIN: https://github.com/Homebrew/brew
HEAD: 556b8f16c6f8518404330780d153fd245b801325
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/linuxbrew-core
Core tap HEAD: 5be975b73353b72bd80d82e60604fd0a32409cd3
Core tap last commit: 2 days ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 2
HOMEBREW_TEMP: /home/linuxbrew/tmp
Homebrew Ruby: 2.6.3 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: dual-core 64-bit sandybridge
Clang: 13.0.0
Git: 2.27.0 => /bin/git
Curl: 7.61.1 => /bin/curl
Kernel: Linux 4.18.0-305.19.1.el8_4.x86_64 x86_64 GNU/Linux
OS: AlmaLinux release 8.4 (Electric Cheetah)
Host glibc: 2.28
/usr/bin/gcc: 8.4.1
/usr/bin/ruby: 2.5.9
glibc: N/A
gcc@5: N/A
xorg: N/A

**Comment by cho-m** [comment, marked as answer]
`GLIBCXX_3.4.29` usually means GCC-11 (e.g. Homebrew/Linuxbrew's `gcc` formula) was used/linked during build.

The error is specifically complaining about `srt` formula:
```
required by /home/linuxbrew/.linuxbrew/lib/libsrt.so.1.4
```

[Linuxbrew's formula for `srt`](https://github.com/Homebrew/linuxbrew-core/blob/master/Formula/srt.rb) should use Ubuntu 16.04 container GCC-5 to build bottle, so it shouldn't reference GCC-11 libs.

---

Maybe check `brew info srt` to see how it was installed (was it poured from bottle, built from source, etc.)?

Then, maybe try reinstalling `srt`.

You can also check the linkage via `objdump -p`.
For example, I just tried poured bottle from Homebrew/core and see:
```console
# objdump -p /home/linuxbrew/.linuxbrew/lib/libsrt.so.1.4 | grep GLIBCXX
    0x0297f870 0x00 20 GLIBCXX_3.4.20
    0x0297f861 0x00 18 GLIBCXX_3.4.11
    0x02297f89 0x00 13 GLIBCXX_3.4.9
    0x0297f865 0x00 07 GLIBCXX_3.4.15
    0x0297f871 0x00 05 GLIBCXX_3.4.21
    0x08922974 0x00 03 GLIBCXX_3.4
```

**Reply by jintor** [reply]
I think `brew reinstall srt` did the trick...

`Checking for dependents of upgraded formulae...`
`Reinstalling 1 broken dependent from source:`
`mlt`
